platforms/shared/configuration/roles/setup/edge-stack/tasks/main.yaml

```diff
@@ -39,12 +33,184 @@
+
+- name: Create custom values for aes helm chart
+  vars:
+    ports: "{{ network.env.ambassadorPorts.ports | default([]) }}"
+    lbSourceRangeDefault:
+      - 0.0.0.0/0
+    loadBalancerSourceRanges: "{{ network.env.loadBalancerSourceRanges | default(lbSourceRangeDefault) }}"
+    elastic_ip: "{{ allocation_ips_stdout }}"
+    extracted_ports: "{{ ambassadorports_list }}"
+  template:
+    src={{ playbook_dir }}/../../../platforms/shared/configuration/roles/setup/edge-stack/templates/aes-custom-values.tpl
+    dest={{ playbook_dir }}/../../../platforms/shared/configuration/build/aes-custom-values.yaml
+  when:
+    - network.type == 'indy'
+    
+- name: Format ambassador range
+  args:
+    executable: /bin/bash
+  shell: |
+    from='{{ network.env.ambassadorPorts.portRange.from | default('') }}'
+    to='{{ network.env.ambassadorPorts.portRange.to | default('') }}'
+    if [ -z "$from" ] || [ -z "$to" ] 
+    then
+      echo ""
+    else
+      echo "--set ambassador.otherPorts.portRange.from=${from} --set ambassador.otherPorts.portRange.to=${to}"
+    fi
+  register: ambassadorRange
+
+
+- name: Format ambassador ports for Indy
+  args:
+    executable: /bin/bash
+  shell: |
+    ports=$(echo '{{ network.env.ambassadorPorts.ports | default('') }}' | sed -e 's/\[/\{/' -e 's/\]/\}/')
+    terminalPorts='{{ terminal.stdout | default('') }}'
+    if [ -z "$ports" ] 
+    then
+      echo ""
+    else
+      echo "--set ambassador.otherPorts.ports={'${ports},${terminalPorts}'}"
+    fi
+  register: ambassadorPortsIndy
+  when:
+    - network.type == 'indy'
+
+- name: check required ports available for Indy
+  args:
+    executable: /bin/bash
+  shell: |
+    availablePortRange=$(echo '{{ ports }}' | sed -e 's/\[/\(/' -e 's/\]/\)/' | sed 's/,/\n/g')
+    from=$(echo '{{ network.env.ambassadorPorts.ports | default('') }}' |sed -e 's/\[/\(/' -e 's/\]/\)/' | sed 's/,/\n/g')
+    to='{{ terminal.stdout | default('') }}'
+
+    if [[ "$from" == "" || "$to" == "" ]]; then
+       echo "false" && exit 0
+    fi
+    arr=( $(seq $from $to) )
+
+    for i in "${arr[@]}"
+    do
+      if [[ ! ${availablePortRange[*]} =~ ${i} ]]; then
+        echo "false" && exit 0
+      fi
+    done
+    echo "true"
+  register: indy_ports_available
+  when:
+    - network.type == 'indy'
+    - ambassador_installed
 
 # Adding the datawire chart repo
 - name: Add datawire chart repo
   kubernetes.core.helm_repository:
     name: datawire
-    repo_url: "https://app.getambassador.io"
+    repo_url: "https://app.getambassador.io" 
+
+# Install edge-stack for indy (NLB) via helmchart
+- name: Deploy the edge-stack
+  kubernetes.core.helm:
+    kubeconfig: "{{ kubeconfig_path }}"
+    name: "edge-stack"
+    chart_ref: datawire/edge-stack
+    chart_version: 8.7.2
+    release_namespace: "{{ proxy_namespace }}"
+    create_namespace: true
+    values:
+      namespace: "{{ proxy_namespace }}"
+      ambassador_Range: "{{ ambassadorRange.stdout }}"
```

Comment:
ambassador_Range and extracted_ports what are these values for ? are they even defined in the aes chart?

Reply:
updated files to update in the custom values file